cmd/podman/containers/create.go

```diff
@@ -292,11 +292,11 @@ func createPodIfNecessary(s *specgen.SpecGenerator) (*entities.PodCreateReport,
 		return nil, errors.Errorf("new pod name must be at least one character")
 	}
 	createOptions := entities.PodCreateOptions{
-		Name:  podName,
-		Infra: true,
-		Net: &entities.NetOptions{
-			PublishPorts: s.PortMappings,
-		},
+		Name:          podName,
+		Infra:         true,
+		Net:           netOpts,
+		Share:         strings.Split(specgen.DefaultKernelNamespaces, ","),
```

Comment:
It looks like we need to make sure that the defaults are used in Specgen if this is unset.

Reply:
[there?](https://github.com/containers/podman/blob/c66ce8d67f9eb53a0997da39999ed8629d3252e3/pkg/specgen/generate/namespaces.go#L462)

Reply:
Yes - that looks right. If the argument to that is nil, we should return the defaults. A 0-length non-nil array would indicate nothing is shared.

Reply:
OK, updated.